WordPress/gutenberg
# [Mobile] Improve List block onSplit to avoid hiding the keyboard on enter

## What?
Fixes https://github.com/WordPress/gutenberg/issues/57580

## Why?
To avoid hiding and showing the keyboard whenever a new item is added to the list.

## How?
In a recent PR https://github.com/WordPress/gutenberg/pull/54543 the Web editor unified the split logic of blocks, after some investigation I confirmed using this new functionality solved the issue with the List block.

For now, it only adapts this feature for this block but we should explore enabling this for all blocks as the web editor did. 

## Testing Instructions

> [!NOTE]  
> Please use the following builds to test:
> - [iOS](https://github.com/wordpress-mobile/WordPress-iOS/pull/23336#issuecomment-2158582869)
> - [Android](https://github.com/wordpress-mobile/WordPress-Android/pull/20962#issuecomment-2158573444)

- Open the editor
- Add a List block
- Add a few items
- **Expect** the keyboard to stay visible when splitting and merging list items

**Note**: Nesting lists make the keyboard hide and show again as the block is recreated.

### Testing Instructions for Keyboard
N/A

## Screenshots or screencast 

iOS Before|iOS After
-|-
<video src="https://github.com/WordPress/gutenberg/assets/4885740/ff15274c-1e28-420e-a90a-6f9762189577" width=250 />|<video src="https://github.com/WordPress/gutenberg/assets/4885740/85aab168-f460-4a98-8559-e7e26471e905" width=250 />
